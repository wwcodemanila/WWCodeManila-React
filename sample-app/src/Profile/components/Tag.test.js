import React from "react";
import { shallow } from 'enzyme';

import Tag from './Tag'

describe('Tag Component', () => {
  test('Tag renders correctly', () => {
    // Snapshot test case
    const component = shallow(<Tag />)
    expect(component).toMatchSnapshot()
  })

  test('Test the click function', () => {
    // Testing Methods from components directly
    const component = shallow(<Tag />)
    const componentInstance = component.instance()
    componentInstance.addLike()
    

    expect(component.state('likes')).toBe(1)
  })
})
