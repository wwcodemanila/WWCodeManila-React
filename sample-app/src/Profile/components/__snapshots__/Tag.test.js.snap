// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component Tag renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="control"
      >
        <div
          className="tags has-addons"
          onClick={[Function]}
        >
          <span
            className="tag is-info"
          >
            Likes
          </span>
          <span
            className="tag"
          >
            0
          </span>
        </div>
      </div>,
      "className": "field is-grouped is-grouped-multiline",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tags has-addons"
          onClick={[Function]}
        >
          <span
            className="tag is-info"
          >
            Likes
          </span>
          <span
            className="tag"
          >
            0
          </span>
        </div>,
        "className": "control",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="tag is-info"
            >
              Likes
            </span>,
            <span
              className="tag"
            >
              0
            </span>,
          ],
          "className": "tags has-addons",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Likes",
              "className": "tag is-info",
            },
            "ref": null,
            "rendered": "Likes",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "tag",
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="control"
        >
          <div
            className="tags has-addons"
            onClick={[Function]}
          >
            <span
              className="tag is-info"
            >
              Likes
            </span>
            <span
              className="tag"
            >
              0
            </span>
          </div>
        </div>,
        "className": "field is-grouped is-grouped-multiline",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="tags has-addons"
            onClick={[Function]}
          >
            <span
              className="tag is-info"
            >
              Likes
            </span>
            <span
              className="tag"
            >
              0
            </span>
          </div>,
          "className": "control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="tag is-info"
              >
                Likes
              </span>,
              <span
                className="tag"
              >
                0
              </span>,
            ],
            "className": "tags has-addons",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Likes",
                "className": "tag is-info",
              },
              "ref": null,
              "rendered": "Likes",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "tag",
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
